ravif/src/dirtyalpha.rs

```diff
@@ -1,66 +1,85 @@
-use imgref::{Img, ImgRef};
-use rgb::{ComponentMap, RGB, RGBA8};
+use imgref::Img;
+use imgref::ImgRef;
+use rav1e::Pixel;
+use rgb::ComponentMap;
+use rgb::Rgb;
+use rgb::Rgba;
 
 #[inline]
-fn weighed_pixel(px: RGBA8) -> (u16, RGB<u32>) {
-    if px.a == 0 {
-        return (0, RGB::new(0, 0, 0));
+fn weighed_pixel<P: Pixel + Default>(px: Rgba<P>) -> (P, Rgb<P>) {
```

Comment:
Weight can't be the same type as the pixel, since it's the pixel value multiplied by weight, so it needs to be wider.

Reply:
Ah yes, this was from the [issue](https://github.com/kornelski/cavif-rs/issues/95) I opened. I didn't know this was intended since this was not documented. I will add some comments and modify the types here.